[FEATURE] Add documentation for new task enhanceManifest and processor manifestEnhancer (#958)
SAP/ui5-tooling: docs/pages/Builder.md

| Task                           | Type `application` | Type `library` | Type `theme-library` |
| ------------------------------ | :----------------: | :------------: | :------------------: |

> | enhanceManifest                | *enabled*          | *enabled*      |                      |

Reviewer: Is this the actual order of execution?

Author: Nope. I thought its sorted alphabetical but its not. So, I have moved the newly added row to the correct place.

??? info "Minification Activity Diagram"
    ![minify Task Activity](../images/UI5_Tooling/Task_Minify.svg){ loading=lazy }

### Generation of supported locales

> The `enhanceManifest` task fills the `supportedLocales` property in the `manifest.json` of a UI5 library/application automatically with the available locales determined by the existence of the respective `.properties` translation file.

Reviewer: The minimum "_version" should be mentioned

Reviewer: It should also be mentioned that this behavior won't apply for files outside of the namespace of the manifest.json.

Author: done

#### Scenario: Application

````txt
- webapp/i18n/
  - i18n.properties
  - i18n_en.properties
  - i18n_en_US.properties
  - i18n_de.properties
  - i18n_de_DE.properties 
````

In the `manifest.json` the `supportedLocales` property will be enhanced as following:

**Source**
`````json
"models": {
  "i18n": {
    "type": "sap.ui.model.resource.ResourceModel",
    "settings": {
	    "bundleName": "sap.ui.demo.todo.i18n.i18n"
    }
 }
`````

**Build Result**
`````json
"models": {
  "i18n": {
    "type": "sap.ui.model.resource.ResourceModel",
    "settings": {
	    "bundleName": "my.app.i18n.i18n"
	    "supportedLocales": [
                 "",
                  "de",
                  "de_DE",
                  "en",
                  "en_US"
	    ]
    }
 }
`````

> To disable the automatic generation of the `supportedLocales`, set `supportedLocales` to any desired value.

Reviewer: The meaning of an empty array should also be explained here.

Author: I would rather add a link to https://ui5.sap.com/#/topic/ec753bc539d748f689e3ac814e129563 instead of explaining runtime code here.

??? info "Minification Activity Diagram"
    ![minify Task Activity](../images/UI5_Tooling/Task_Minify.svg){ loading=lazy }

### Generation of supported locales

The `enhanceManifest` task fills the `supportedLocales` property in the `manifest.json` of a UI5 library/application automatically with the available locales determined by the existence of the respective `.properties` translation file. To disable the automatic generation of the `supportedLocales`, set `supportedLocales` to any desired value. For further resource bundle configuration options, see [Supported Locales and Fallback Chain](https://ui5.sap.com/#/topic/ec753bc539d748f689e3ac814e129563).

#### Requirements

> This feature is only available for projects:

Reviewer 2: ```suggestion
This feature is only available for the following projects:
```

> This feature is only available for the following projects:

Author: done

> - having set the `_version` property in the `manifest.json` to `1.21.0` or newer

Reviewer 2: ```suggestion
- The `_version` property in the `manifest.json` is set to `1.21.0` or higher
```

> - the defined bundle is located inside the namespace of the `manifest.json`

Reviewer 2: ```suggestion
- The defined bundle is located inside the namespace of the `manifest.json`
```

> - The defined bundle is located inside the namespace of the `manifest.json`

Author: done

#### Scenario: Application

````txt
- webapp/i18n/
  - i18n.properties
  - i18n_en.properties
  - i18n_en_US.properties
  - i18n_de.properties
  - i18n_de_DE.properties 
````

In the `manifest.json` the `supportedLocales` property will be enhanced as follows:

**Source**

> `````json

Author: done

"models": {
  "i18n": {
    "type": "sap.ui.model.resource.ResourceModel",
    "settings": {
	    "bundleName": "my.app.i18n.i18n"
    }
 }

> `````

Author: done

**Build Result**

> `````json

Author: done

>  }

Reviewer 3: ```suggestion
  }
}
```

Sorry for the nitpick 😄

Author: you're right.... I even added another bracket

"models": {
  "i18n": {
    "type": "sap.ui.model.resource.ResourceModel",
    "settings": {
	    "bundleName": "my.app.i18n.i18n",
	    "supportedLocales": [
                 "",
                  "de",
                  "de_DE",
                  "en",
                  "en_US"
	    ]
    }
 }

> `````

Author: done

> `````

Author: done

>  }

Reviewer 3: ```suggestion
  }
}
```

Author: you're right.... I even added another bracket
